disperser/common/blobstore/blob_metadata_store.go

```diff
@@ -130,6 +130,9 @@ func (s *BlobMetadataStore) GetBlobMetadataByStatusCount(ctx context.Context, st
 
 // GetBlobMetadataByStatusWithPagination returns all the metadata with the given status upto the specified limit
 // along with items, also returns a pagination token that can be used to fetch the next set of items
+//
+// Note that this may not return all the metadata for the batch if dynamodb query limit is reached.
+// e.g 1mb limit for a single query
```

Comment:
How is this enforced?

Reply:
From the dynamodb docs https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html#API_Query_RequestSyntax

> A single Query operation will read up to the maximum number of items set (if using the Limit parameter) or a maximum of 1 MB of data